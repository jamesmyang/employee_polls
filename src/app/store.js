import { configureStore } from '@reduxjs/toolkit'

import usersReducer from '../features/users/usersSlice'
import questionsReducer from '../features/questions/questionsSlice'

export default configureStore({
  reducer: {
    users: usersReducer,
    questions: questionsReducer,

  }
})
