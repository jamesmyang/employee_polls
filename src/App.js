import React from 'react'
import { useSelector } from 'react-redux'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom'

import { Navbar } from './app/Navbar'
import { LoginPage } from './features/users/LoginPage'
import { QuestionsPage } from './features/questions/QuestionsPage'
import { SingleQuestionPage } from './features/questions/SingleQuestionPage'

function App() {
  const userId = useSelector(state => state.users.auth.userId)
  const isLoggedIn = Boolean(userId)

  return (
    <Router>
      {!isLoggedIn ? (
        <Switch>
          <Route
            path="/login"
            component={LoginPage} />
          <Redirect to="/login" />
        </Switch>
      ) : (
        <div>
          <Navbar />
          <div className="App">
            <Switch>
              <Route
                exact
                path="/"
                component={QuestionsPage} />
              <Route
                exact
                path="/questions/:questionId"
                component={SingleQuestionPage} />
              <Redirect to="/" />
            </Switch>
          </div>
        </div>
      )}
    </Router>
  )
}

export default App
