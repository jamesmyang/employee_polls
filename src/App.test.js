import * as React from 'react';
import { Provider } from 'react-redux'
import { render } from '@testing-library/react';
import App from './App';
import store from './app/store'

describe('App', () => {
  it('matches the snapshot for App', async () => {
    var component = render(
      <Provider store={store}>
        <App />
      </Provider>
    )

    expect(component).toMatchSnapshot();
  })


})