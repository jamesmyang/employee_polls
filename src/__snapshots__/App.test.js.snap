// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot for App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="center"
        style="width: 80%;"
      >
        <img
          alt="EmployeePolls Logo"
          src="/employeepolls.png"
        />
        <h4>
          Log in
        </h4>
        <div />
        <form>
          <label
            for="userId"
          >
            User
          </label>
          <select
            id="userId"
            name="userId"
          >
            <option
              value=""
            />
          </select>
          <label
            for="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
            type="password"
            value=""
          />
          <button
            disabled=""
            type="button"
          >
            Submit
          </button>
        </form>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="center"
      style="width: 80%;"
    >
      <img
        alt="EmployeePolls Logo"
        src="/employeepolls.png"
      />
      <h4>
        Log in
      </h4>
      <div />
      <form>
        <label
          for="userId"
        >
          User
        </label>
        <select
          id="userId"
          name="userId"
        >
          <option
            value=""
          />
        </select>
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
        <button
          disabled=""
          type="button"
        >
          Submit
        </button>
      </form>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
